lib/tlog/es_json_reader.c

```diff
@@ -380,12 +381,19 @@ tlog_es_json_reader_refill_array(struct tlog_es_json_reader *es_json_reader)
 
     /* Perform the request */
     rc = curl_easy_perform(es_json_reader->curl);
+
+    /* Check the HTTP response */
+    curl_easy_getinfo(es_json_reader->curl, CURLINFO_RESPONSE_CODE, &http_status);
+
     if (rc != CURLE_OK) {
         if (rc == CURLE_WRITE_ERROR && data.rc != json_tokener_success) {
             grc = TLOG_GRC_FROM(json, data.rc);
         } else {
             grc = TLOG_GRC_FROM(curl, rc);
         }
+        if (http_status > 0 && http_status != 200) {
+            fprintf(stderr, "Failed HTTP request to %s (%ld response)", url, http_status);
+        }
```

Comment:
I would prefer to provide a libcurl option like `curl -i` to print the HTTP status response, then let the Administrator decide how to parse that information.

Reply:
OK, I removed this bit. It was kind of useful because tlog-play was trying to parse the data, whatever the response, so you got a "write error" back from curl which is kind of confusing. I set FAILONERROR, which means it now tells you the server responded with an error.

It doesn't tell you what the error is, but the --es-verbose option now provides what is needed.